velox/functions/prestosql/ArraySort.cpp

```diff
@@ -57,29 +57,37 @@ BufferPtr sortElements(
     for (auto i = offset; i < offset + size; ++i) {
       rawIndices[i] = i;
     }
-    std::sort(
-        rawIndices + offset,
-        rawIndices + offset + size,
-        [&](vector_size_t& a, vector_size_t& b) {
-          bool aNull = decodedElements->isNullAt(a);
-          bool bNull = decodedElements->isNullAt(b);
-          if (aNull) {
-            return false;
-          }
-          if (bNull) {
-            return true;
-          }
-
-          std::optional<int32_t> result = baseElementsVector->compare(
-              baseElementsVector, decodedIndices[a], decodedIndices[b], flags);
-
-          VELOX_USER_CHECK(
-              result.has_value(),
-              "{} comparison not supported for values that contain nulls",
-              mapTypeKindToName(baseElementsVector->typeKind()));
-
-          return result.value() < 0;
-        });
+
+    try {
+      std::sort(
+          rawIndices + offset,
+          rawIndices + offset + size,
+          [&](vector_size_t& a, vector_size_t& b) {
+            bool aNull = decodedElements->isNullAt(a);
+            bool bNull = decodedElements->isNullAt(b);
+            if (aNull) {
+              return false;
+            }
+            if (bNull) {
+              return true;
+            }
+
+            std::optional<int32_t> result = baseElementsVector->compare(
+                baseElementsVector,
+                decodedIndices[a],
+                decodedIndices[b],
+                flags);
+
+            if (!result.has_value()) {
+              VELOX_USER_FAIL(
+                  "array_sort contains nested nulls not supported for comparison");
+            }
+
+            return result.value() < 0;
+          });
+    } catch (const VeloxUserError& exception) {
```

Comment:
You can achieve the same by using rows.applyToSelectedNoThrow. Please, go over the links I shared earlier.